File: Phan_A/bai36/bai_36.py
import sys

def giai_thua(n: int) -> int:
    kq = 1
    for i in range(1, n + 1):
        kq *= i
    return kq

if __name__ == "__main__":
    input_data = sys.stdin.read().split()
    if input_data:
        n = int(input_data[0])
        # Nguong gioi han LLONG_MAX cua C++ de mo phỏng kiem tra tran so
        LLONG_MAX = 9223372036854775807
        kq = giai_thua(n)
        if kq > LLONG_MAX:
            print(f"TRAN SO ({n}! = {kq} > {LLONG_MAX})")
        else:
            print(kq)

File: Phan_A/bai36/bai_36.jl
function giai_thua(n::Int)
    LLONG_MAX = 9223372036854775807
    kq = BigInt(1)
    for i in 1:n
        kq *= i
    end
    if kq > LLONG_MAX
        # Dùng $(n)! để Julia không bị nhầm lẫn với hàm n!
        println("TRAN SO ($(n)! = $kq > $LLONG_MAX)")
    else
        println(Int(kq))
    end
end

input_data = split(read(stdin, String))
if !isempty(input_data)
    n = parse(Int, input_data[1])
    giai_thua(n)
end